llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -278,11 +281,16 @@ static void copyRegOperand(MachineOperand &To, const MachineOperand &From) {
   }
 }
 
+static bool isSameReg(const MachineOperand &Op, Register Reg) {
+  return Op.isReg() && Op.getReg() == Reg;
```

Comment:
isReg can never be false for the copy-like contexts you're looking at

Reply:
I have removed the `isSameReg` overloads that I had previously introduced and replaced the use of this one by `Op.getReg() == Reg`.